### PR TITLE
Exact Frobenius Init with Tangent-Only U/W Floor on PR294 Radial Brake (3.28 in 3010 with 3 seeds)

### PR DESCRIPTION
**Pull request by SPThole**
This is a three-seed Track 3 follow-up on top of PR #294's radial damping
optimizer stack. Relative to PR #294, it changes both the `u/w` floor behavior
and the hidden-matrix initialization. The benchmark dataset, batch size, model
architecture, and training endpoint are unchanged.

## What Changed

- Keeps PR #294 outward radial damping and post-step radius correction.
- Changes PR #294's full-update `u/w` floor to the C-style tangent-only `u/w`
  floor: the floor scales only tangent update energy after radial damping,
  leaving the dampened radial component intact.
- Initializes non-projection hidden matrices with
  `std = sqrt(0.33 / fan_in)`, then rescales each matrix to its exact target
  Frobenius norm.
- Affected initialized matrices: `blocks.*.attn.{q,k,v}.weight` and
  `blocks.*.mlp.fc.weight`.
- Projection parameters remain zero-initialized by name, and all non-projection
  biases are zeroed.
- Adds denser near-end validation checkpoints:
  `2960, 2970, 2975, 2980, 2985, 2990, 2995, 2999, 3000, 3010, 3020`.

## Result and Z Score

Three seeds were run: `0`, `1`, and `2`. No seed was dropped.

At step `3010`, the mean validation loss is `3.27748333`:

```text
(3.28 - 3.27748333) * sqrt(3) = 0.00435899 >= 0.004
z = 3.3531
```

At the final step `3020`, the mean validation loss is `3.27683000`:

```text
(3.28 - 3.27683000) * sqrt(3) = 0.00549060
z = 4.2235
```

## Seed-Level Convergence

| Seed | 2960 | 2970 | 2975 | 2980 | 2985 | 2990 | 2995 | 2999 | 3000 | 3010 | 3020 |
| -: | -: | -: | -: | -: | -: | -: | -: | -: | -: | -: | -: |
| 0 | 3.28025 | 3.27954 | 3.27918 | 3.27885 | 3.27850 | 3.27814 | 3.27782 | 3.27757 | 3.27750 | 3.27684 | 3.27621 |
| 1 | 3.28080 | 3.28008 | 3.27975 | 3.27941 | 3.27903 | 3.27868 | 3.27836 | 3.27813 | 3.27805 | 3.27736 | 3.27670 |
| 2 | 3.28165 | 3.28092 | 3.28058 | 3.28025 | 3.27987 | 3.27952 | 3.27924 | 3.27903 | 3.27895 | 3.27825 | 3.27758 |
| **Mean** | **3.28090000** | **3.28018000** | **3.27983667** | **3.27950333** | **3.27913333** | **3.27878000** | **3.27847333** | **3.27824333** | **3.27816667** | **3.27748333** | **3.27683000** |

Same-seed comparison versus PR #294 logs for seeds `0`, `1`, and `2`
(`PR #294 val - this run val`; positive means this run is lower):

| Step | Seed 0 | Seed 1 | Seed 2 | Mean |
| -: | -: | -: | -: | -: |
| 2875 | +0.00152 | +0.00120 | -0.00146 | +0.00042 |
| 2990 | +0.00149 | +0.00128 | -0.00158 | +0.00040 |
| 2995 | +0.00149 | +0.00131 | -0.00162 | +0.00039 |
| 2999 | +0.00150 | +0.00129 | -0.00164 | +0.00038 |
| 3000 | +0.00150 | +0.00130 | -0.00162 | +0.00039 |
| 3010 | +0.00148 | +0.00130 | -0.00161 | +0.00039 |
| 3020 | +0.00145 | +0.00130 | -0.00158 | +0.00039 |

## Note

Because of limited H100 compute (GPU poor 🫠 ), this record only runs three seeds. Although PR #294 remains the official 11-seed SOTA at 2990 steps, on the matched seed subset 0,1,2, this experiment clears the Track 3 precision condition 10 steps earlier than PR #294’s corresponding seeds: 3010 vs 3020. The same-seed tail mean is about 0.00039 lower, roughly equivalent to 8-10 steps under the Track 3 loss-per-step heuristic.

## Lineage

Built on PR #294 by @nilin, with inherited components from PR #291, PR #287
PowerCool scheduling, PR #278 SOAP-style preconditioning, PR #275 Contra-Muon,
and PR #274 Skylight/NorMuon-lite.

### CODE REVIEW

**Comment by KellerJordan** [comment]
hello, thank you for the result submission. hm, seems like we don't quite have stat sig on what difference this is making. I'm also not sure I would accept if proposed delta is marginally worse.

**Comment by SPThole** [comment]
Thanks for the feedback @KellerJordan ,,  I completely understand the concern regarding statistical significance, especially given the limited seed count.
To clarify the intent: the submission was based on a direct, same-seed comparison against PR #294. While it looks like a marginal difference globally, when isolating the exact same seed combinations (0, 1, and 2), this configuration clears the Track 3 precision threshold 10 steps earlier (3010 vs. 3020 steps) with a tail mean that is consistently ~0.00039 lower.
The variance is likely driven by the exact Frobenius norm initialization. However, since I was constrained by H100 compute, I couldn't scale this up to the full 11-seed suite to definitively prove a robust global improvement. Completely understand if we want to hold off due to the small sample size!
